ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml

```diff
@@ -151,12 +151,26 @@ objects:
           break
         done
 
+        function isRouterReady() {
+          local deployment
+          deployment=${1:-"router/deploy"}
+          local ns
+          ns=${2:-"openshift-ingress"}
+          if [[ "${deployment}" == "ds/router-default" ]]; then
+            ready=$(oc get "${deployment}" -n "${ns}" -o go-template='{{ne "0" (print .status.numberReady)}}')
+            [[ "${ready}" == "true" ]]
+            return
+          fi
+
+          oc wait "${deployment}" -n "${ns}" --for condition=available --timeout=10m
```

Comment:
Does this `--timeout` cause each iteration to wait 10 minutes *in addition to* the 60 seconds in the loop?!

Reply:
that was the original wait code,  it was `(oc wait 10 mins + 1 min sleep)  * 10 retries`.

Reply:
I wonder why? Seems like a good way to waste like 10 minutes during setup... it takes a matter of seconds for the router to deploy

Reply:
We were looping waiting because of a bug in wait.

Reply:
[Here][1] is a sketch of something that may be more robust for looping on blocking calls.

[1]: https://github.com/openshift/release/pull/2004/files#r227924803